### PR TITLE
gnmic: add page and subpages

### PR DESCRIPTION
**Pull request by alexandrehassan**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **v0.26.0**

### CODE REVIEW

File: pages/common/gnmic-get.md
Comment (CleanMachine1) [inline pages/common/gnmic-get.md:12]:
```suggestion
`gnmic -a {{ip}}:{{port}} get --path {{path1}} --path {{path2}}`
```

File: pages/common/gnmic-get.md
Comment (CleanMachine1) [inline pages/common/gnmic-get.md:16]:
```suggestion
`gnmic -a {{ip}}:{{port}} get --prefix {{prefix}} --path {{path1}} --path {{path2}}`
```

**Review by CleanMachine1 — COMMENTED** [review]
could you apply this throughout

File: pages/common/gnmic.md
Comment (adrienthebo) [inline pages/common/gnmic.md:4]: Taking the perspective of someone that's not familiar with `gnmic` or gNMI more broadly, I'd want this description to explain what `gnmic` is used for and why I might use it. This description indicates what gnmic is capable of but if I'm a new user that just encountered this command in a script and want to figure out what it's doing, I won't gain that understanding beyond knowing that I need to start searching for "gNMI".

This [Cisco gNMI Protocol](https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/prog/configuration/168/b_168_programmability_cg/gNMI_protocol.pdf) document indicates that gNMI is used to interact with/set configuration on network devices. Would the following be an accurate description of what gnmic is used for?

```suggestion
> A gNMI command-line client.
> Manage gNMI network device configuration and view operational data.
> More information: <https://gnmic.kmrd.dev>.
```

File: pages/common/gnmic.md
Comment (adrienthebo) [inline pages/common/gnmic.md:10]: Command examples typically describe what a command is doing and how it's used, can we indicate 

```suggestion
- Provide a username and password to fetch device capabilities:
```

File: pages/common/gnmic.md
Comment (adrienthebo) [inline pages/common/gnmic.md:6]:
```suggestion
- Request device capabilities:
```

File: pages/common/gnmic.md
Comment (adrienthebo) [inline pages/common/gnmic.md:14]:
```suggestion
- Get a snapshot of the device state at {{path}}:
```

File: pages/common/gnmic.md
Comment (adrienthebo) [inline pages/common/gnmic.md:24]: Going off of the [gNMI reference section 3.5](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#35-subscribing-to-telemetry-updates):

```suggestion
- Subscribe to target state updates under the subtree {{path}}:

`gnmic -a {{ip:port}} --insecure subscribe --path {{path}}`
```

File: pages/common/gnmic.md
Comment (adrienthebo) [inline pages/common/gnmic.md:18]: Going off if the [gNMI specification section 3.4](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#34-modifying-state)

```suggestion
- Update device state at {{path}}:
```

**Review by adrienthebo — COMMENTED** [review]
Hi @alexandrehassan, thanks for this submission! I've reviewed `pages/common/gnmic.md` with some preliminary thoughts and suggestions.

In broad terms, as someone whose first introduction to gNMI and `gnmic` is this pull request, I'm looking for things like what commands I should know right away, and how things are intended to be used.

Descriptions like this

```
> Set RPC allows the client to modify the state of data on the target. The data specified referenced by a path can be updated, replaced or deleted.
```

are really helpful because I understand that this is manipulating data on a (gNMI device) target and there are different ways for me to manipulate data. Descriptions like this:

```
 > It is used to send a GetRequest to the specified target(s) (using the global flag --address and expects one GetResponse per target, per path.
```

tell me that I'm sending a GetRequest, but the description carries more information about the structure of the RPC protocol and less about the fact that `gnmic get` is fetching state from a network device.

The examples are thorough but a lot of the material assumes familiarity with `gNMI` and how it's used. The [snmpwalk](https://github.com/tldr-pages/tldr/blob/main/pages/linux/snmpwalk.md) documentation has a similar case of interacting with data on a remote system; could we amend these examples with this in mind?

**Comment by alexandrehassan** [comment]
@adrienthebo thanks a lot for the feedback, I have tried modifying the content to be more in line with your suggestions and have added to more examples that ended up being useful (encoding makes piping to a file super easy). Would love to have your feedback

File: pages/common/gnmic-subscribe.md
Comment (adrienthebo) [inline pages/common/gnmic-subscribe.md:12]:
```suggestion
`gnmic -a {{ip:port}} subscribe --path {{path}} --sample-interval 30s`
```

File: pages/common/gnmic-get.md
Comment (adrienthebo) [inline pages/common/gnmic-get.md:14]:
```suggestion
- Query the device state at multiple paths with a common prefix ({{prefix}}/{{path}}):
```

File: pages/common/gnmic-get.md
Comment (adrienthebo) [inline pages/common/gnmic-get.md:18]:
```suggestion
- Query the device state and specify reponse encoding (json_ietf):
```

File: pages/common/gnmic.md
Comment (adrienthebo) [inline pages/common/gnmic.md:5]: We may want to draw attention that there are tldr pages for subcommands, but let's get a 👍 / 👎  some of the other tldr collaborators before proceeding with this change. Thoughts?

```suggestion
> Some subcommands such as `get`, `set`, and `subscribe` have their own usage documentation, accessible via `tldr gnmic-subcommand`.

> More information: <https://gnmic.kmrd.dev>.
```

File: pages/common/gnmic.md
Comment (adrienthebo) [inline pages/common/gnmic.md:9]: Question: we use `--insecure` as a flag for all of these examples but don't use them in other tldr pages. Is it worthwhile to have a single example explaining why this flag would be used, and then drop the flag from other examples?

**Review by adrienthebo — COMMENTED** [review]
I've a few small fixup suggestions, and there are some points that I'd like to get another tldr collaborator to double check a solution. Other than that this is coming together nicely!

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic-get.md:6]:
```suggestion
- Get a snapshot of the device state at specific path:
```

It is not mentioned in our style guide what to do with placeholders inside descriptions (but it should be). Anyway I think we should not use them anywhere outside code examples.

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic-get.md:8]:
```suggestion
`gnmic --address {{ip:port}} get --path {{path}}`
```

Long options are preferred. But now there are no fixed rules when we use them. So you for instance can tell me that this option is frequently used or i extremely portable and I dismiss this comment.

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic-get.md:12]:
```suggestion
`gnmic --aaddress {{ip:port}} get --path {{path1}} --path {{path2}}`
```

For paths we use `{{path/to/directory}}`, `{{path/to/file}}` generic placeholders. I don't know semantics of this path so please decide how better to fix it yourself. ;)

Also it's better to provide a concrete value instead of `{{ip:port}}` here because `--address` describes the meaning of the following argument.

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic-get.md:14]:
```suggestion
- Querry the device state at multiple paths with a common prefix:
```

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic-get.md:16]:
```suggestion
`gnmic --address {{ip:port}} get --prefix {{prefix}} --path {{path1}} --path {{path2}}`
```

The same goes here.

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic-get.md:20]:
```suggestion
`gnmic --address {{ip:port}} get --path {{path}} --encoding json_ietf`
```

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic-get.md:16]:
```suggestion
`gnmic --address {{ip:port}} get --prefix {{prefix}} --path {{path1}} --path {{path2}}`
```

The same goes here.

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic-get.md:18]:
```suggestion
- Query the device state with a specific response encoding:
```

**Review by EmilySeville7cfg — CHANGES_REQUESTED** [review]
Apply the same suggestions to all other pages.

File: pages/common/gnmic-get.md
Comment (alexandrehassan) [reply pages/common/gnmic-get.md]: I have updated the first mentions of `-a` in every page/subpage to follow the long format for clarity but gnmic commands tend to become very long very fast and the address is always included, so I think that using the -a in every other spot is a better choice (once to explain the parameter, the rest is implied)

File: pages/common/gnmic-get.md
Comment (alexandrehassan) [reply pages/common/gnmic-get.md:12]: I'm not sure what to write for the paths, (example paths could be `"/"`, `/`, `info:/`, `info:/network_instances`, `/state/port[port-id=1/1/c1/1]/statistics`, `info:/port[port-id=1/1/c1/1]/statistics`) so I think that just leaving `{{path}}` is ok. 

I'm not sure what is meant for concrete value?

File: pages/common/gnmic.md
Comment (alexandrehassan) [reply pages/common/gnmic.md:9]: I removed the `--insecure` flag mentions.

File: pages/common/gnmic.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic.md:9]: IMO, it's better to provide a specific ip and port examples because `--address` is a long option and describes the meaning of the following placeholder.

File: pages/common/gnmic.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic.md:13]:
```suggestion
`gnmic --address {{ip:port}} --username {{value}} --password {{value}} capabilities`
```

It's better not to repeat when possible.

File: pages/common/gnmic.md
Comment (EmilySeville7cfg) [inline pages/common/gnmic.md:17]:
```suggestion
`gnmic -a {{ip:port}} get --path {{/path/to/device}}`
```

I am doubt about how to write this.

**Review by EmilySeville7cfg — CHANGES_REQUESTED** [review]

File: pages/common/gnmic.md
Comment (alexandrehassan) [reply pages/common/gnmic.md:13]: Sounds good

Comment (EmilySeville7cfg) [reply pages/common/gnmic.md:13]: There is formal rule for this now but it should be. At the end of the day I can mark this conversation as resolved due to the reason mentioned earlier.

Comment (CleanMachine1) [reply pages/common/gnmic.md:13]: When has this been added as a rule?

We should always use informative placeholders

Comment (EmilySeville7cfg) [reply pages/common/gnmic.md:13]: I've missed "no".

**Review by EmilySeville7cfg — APPROVED** [review]

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [reply pages/common/gnmic-get.md]: I think we should formalize when to prefer short options not to write very long commands.

File: pages/common/gnmic-get.md
Comment (EmilySeville7cfg) [reply pages/common/gnmic-get.md:12]: Just leave it as is if you don't know what concrete value to provide. Basically, I don't know too. 🤔

File: pages/common/gnmic.md
Comment (alexandrehassan) [reply pages/common/gnmic.md:9]: I think 127.0.0.1 is the only address that wouldn't work and addresses should be formatted as `address_of_server:port_to_connect`

**Review by adrienthebo — APPROVED** [review]
LGTM, @EmilySeville7cfg since you've touched this last I'll defer to you on merging this (in case anything else needs to get addressed before we merge).

Thanks for the contribution, @alexandrehassan!

**Comment by EmilySeville7cfg** [comment]
Merged, thanks! 😁

**Pull request merged**
